### ISSUE TITLE
Desktop Context Menu

### ISSUE THREAD

**Issue by Mobeeuz** [terrymacdonald/DisplayMagician#77]
I understand why it's there, but it should be optional. I don't see a way to turn it off.

![Context](https://user-images.githubusercontent.com/33134962/148718770-3bdcefa1-94a9-4aec-ac94-36c413529ce3.jpg)

**Comment by terrymacdonald** [comment]
@Mobeeuz why should it be optional? Can you give me some reasons please?  Based on comments I've seen it's quite popular.

It would require me rewriting the installer, which I am contemplating in the future as I'm hitting some limitations with the method being used now. But it's something that would be low on my priority list, mainly because it's only visible when you right click on the desktop. The impact to users is greater I believe to not have it.

TLDR I'll keep it on the list and will review it, and I might add an installer option if I ever redo the installer.

Thanks
Terry

**Comment by Mobeeuz** [comment]
Because personally I don't need it, won't use it, it makes the menu way too fat, and cramming things into peoples system menus without permission is rude ;) Even essential utilities like 7zip make system menu changes optional because despite their obvious benefits, not everyone wants or needs that.

Rather than rewriting the installer, just add a checkbox in the program that would remove it - or reinstate it if desired.

**Comment by terrymacdonald** [comment]
@Mobeeuz We'll have to disagree on the rudeness of helping provide multiple options for accessing the software, but I do understand it is a concern for some. 

It's more work to do it through the software itself as it installation can only be done as an elevated user, and currently DisplayMagician does it's best to avoid asking for admin rights. There is no logic within DisplayMagician to ask for elevated privileges as I want to minimise the security risks as much as I can. Only the installer asks for admin privs, as then it is required anyway in order to install to the Windows Program Files location.

I will keep this issue here and if I get some spare time I will look at adding it to the installer and/or DisplayMagician itself (no promises though). I may look at moving to the register-free COM usage method too so that it makes installation easier, so this may work in well with that work.

Thanks
Terry

**Comment by terrymacdonald** [comment]
ok @Mobeeuz I had a few hours this afternoon to develop at least part of the code needed for your 'remove my desktop context menu' wish. This is now a button in the Program Settings to uninstall the Desktop Context Menu:

![image](https://user-images.githubusercontent.com/5721772/151645756-44204caa-699a-413a-87fb-aefddf4f9ee3.png)

IMPORTANT: When you install a new DisplayMagician version, you will need to manually uninstall the Desktop Context Menu again. This is because the DisplayMagician installer is not smart enough to know that the user doesn't want the Desktop Context Menu installed. 

I will not be updating the current DisplayMagician installer because I plan to replace it with another installation mechanism in the future, so any new DisplayMagician install/update will always reinstall the DisplayMagician Desktop Context Menu again.

Hopefully it's not too much trouble to go in and uninstall it from the program settings each time.

Also be aware that when you press the install/uninstall button, DisplayMagician will restart Windows Explorer so that the Desktop Context Menu change happens immediately.

With all that said an done, here is a test version of DisplayMagician with that functionality in place. Please test it out, and tell me if there are any bugs. NOTE: There are quite a few changes in v2.2.0.112 compared to v2.1.5.5, so you will be asked to recreate new display profiles again. Please make sure you do this so that everything will work ok.
[DisplayMagicianSetup-v2.2.0.112.zip](https://github.com/terrymacdonald/DisplayMagician/files/7962890/DisplayMagicianSetup-v2.2.0.112.zip)

Thanks
Terry

**Comment by Mobeeuz** [comment]
Works beautifully! Adds or removes the option with the click of a button - even more than was asked for.

Suggestion: how about naming the menu "Display Magician Profiles" instead of "DisplayMagician: Change Display Profiles..." for brevity - cuts down on that fat menu size and gets across the same point :)

**Comment by terrymacdonald** [comment]
Fixed in codebase. Will be released with DisplayMagician v2.2.0 whenever that finally gets finished.
